### ISSUE TITLE
Unlocking Spam with Transformers

### ISSUE THREAD

**Issue by mderosa001** [mderosa001/Capstone_Issues#7]
### Title
Unlocking Spam with Transformers: Sentiment-Driven Detection via Fine-tuned BERT

### URL
https://medium.com/@areeshanajam275/from-word-embedding-to-transformer-based-architecture-fine-tuning-bert-for-text-classification-of-80f87906d63c
### Summary 
BERT is a transformer-based NLP model that understands the context of words in a sentence, making it highly effective for complex language tasks. The tutorial covers the entire process, from understanding BERT's architecture and word embeddings to fine-tuning it for sentiment analysis using TensorFlow. It explains data preprocessing, handling class imbalances, training, and evaluating the model. By the end, readers can apply BERT to their own text classification challenges using TensorFlow Hub's pre-trained models.
### Key Points 
BERT Overview: BERT understands language context by analyzing the relationship between words, making it ideal for NLP tasks.
Word Embeddings: BERT creates contextualized word embeddings, overcoming limitations of traditional models like Word2Vec.
Transformer Architecture: BERT uses an encoder-based transformer model for deep contextual understanding.
Spam Detection Example: The tutorial applies BERT to classify emails as spam or ham, addressing data imbalance through oversampling/undersampling.
Using TensorFlow Hub: Pre-trained BERT models from TensorFlow Hub are used for fine-tuning on specific tasks.
Training Process: Steps include loading data, preprocessing text, building and fine-tuning the model, and evaluating its accuracy.
### Citation
@misc{UnlockingSpam,
	author = {AreeshaNajam},
	title = {{U}nlocking {S}pam with {T}ransformers: {S}entiment-{D}riven {D}etection via {F}ine-tuned {B}{E}{R}{T} --- areeshanajam275},
	howpublished = {\url{https://medium.com/@areeshanajam275/from-word-embedding-to-transformer-based-architecture-fine-tuning-bert-for-text-classification-of-80f87906d63c}},
	year = {},
	note = {[Accessed 24-02-2025]},
}
